static/src/js/components/KeywordRecommendations/KeywordRecommendations.jsx

```diff
@@ -0,0 +1,344 @@
+   * @returns the augmented list
+   */
+  // eslint-disable-next-line arrow-body-style
+  const createRecommendedKeywords = (keywordRecommendations) => {
+    return keywordRecommendations.map((recommendedKeyword) => {
+      const { keyword: delimitedKeyword } = recommendedKeyword
+      let accepted = false
+
+      if (isInDraftKeywordList(delimitedKeyword)) {
+        accepted = true
+      }
+
+      const newKeyword = {
+        ...recommendedKeyword,
+        recommended: true,
+        accepted
+      }
+
+      return newKeyword
+    })
+  }
+
+  /**
+   * Store state for component dismount
+   */
+  useEffect(() => {
+    stateRef.current = {
+      draftKeywords,
+      recommendations,
+      requestId
+    }
+  }, [draftKeywords, recommendations, requestId])
+
+  /**
+   * Report GKR feedback.   Sends a list of new, accepted, and rejected keywords.
+   */
+  const reportFeedback = () => {
+    // eslint-disable-next-line no-shadow
+    const { requestId, draftKeywords, recommendations } = stateRef.current
+    if (requestId) {
+      const recommendationObj = {}
+      recommendations.forEach((recommendation) => {
+        const { uuid, accepted } = recommendation
+        recommendationObj[uuid] = accepted
+      })
+
+      const newKeywords = draftKeywords.filter((keyword) => {
+        const { keyword: delimitedKeyword } = keyword
+
+        return !isInRecommendedKeywordList(recommendations, delimitedKeyword)
+      })
+
+      sendKeywordRecommendationsFeedback(
+        requestId,
+        recommendationObj,
+        newKeywords.map((draftKeyword) => (draftKeyword.keyword))
+      )
+    }
+  }
+
+  /**
+   * Retrieves the initial recommended keywords on component mount.
+   */
+  useEffect(() => {
+    const fetchKeywords = async () => {
+      const result = await getKeywordRecommendations(abstract)
+      const { uuid, recommendations: keywordRecommendations } = result
+      const uniqKeywordRecommendations = uniqBy(keywordRecommendations, 'keyword')
+      const keywords = createRecommendedKeywords(uniqKeywordRecommendations)
+      setRecommendations(keywords)
+      setRequestId(uuid)
+      setLoading(false)
+    }
+
+    if (abstract && getDraftKeywords().length === 0) {
+      setLoading(true)
+      fetchKeywords()
+        .catch((error) => {
+          errorLogger('error fetching keywords from GKR', error.message)
+          setLoading(false)
+        })
+    }
+
+    return () => {
+      reportFeedback()
+    }
+  }, [])
+
+  /**
+   * If the science keyword form data changes, update the draft keywords list.
+   */
+  useEffect(() => {
+    const keywords = getDraftKeywords()
+
+    setDraftKeywords(keywords)
+  }, [draft])
+
+  /**
+   * If the science keyword form data changes, update the recommended keywords list.
+   */
+  useEffect(() => {
+    const keywords = createRecommendedKeywords(recommendations)
+    setRecommendations(keywords)
+  }, [draft])
+
+  //
```

Comment:
This comment has an extra line below and above. I see this in quite a few places. We should be using the `//` with no extra lines above and below for inline comments, and the `/* */` jsdoc blocks when describing functions